core/src/main/java/com/mongodb/hadoop/util/MongoConfigUtil.java

```diff
@@ -1127,13 +1154,45 @@ public static void close(final Mongo client) {
 
     private static MongoClient getMongoClient(final MongoClientURI uri) throws UnknownHostException {
         MongoClient mongoClient = CLIENTS.get().get(uri);
-            if (mongoClient == null) {
+        if (mongoClient == null) {
+            // When connction's hostname are IP address, it is parsed into our localhost name  
+            // So it is necessary to pass CA Certification with these following mechanism
+            if (isLocalhost(uri)) {
```

Comment:
@krishnapriawan I am not sure I understand this bit, wouldn't the hostname always be `localhost` since we always establish tunnel connections on clusters for indirect connections?

Can't we just check if SSL is enabled then use `sslContext` if so, like on the previous [code](https://github.com/xplenty/mongo-hadoop/blob/0b9f5684a7cb7bee47a448fd4c08d8a757293a1c/core/src/main/java/com/mongodb/hadoop/util/MongoConfigUtil.java#L1069)?

Reply:
@mlsaito I think if we use previous code implementation, there is a chance that for indirect connection when SSL is disabled then it will fall into else statement which is will be an error. Moreover, if I understand correctly the SSL connection to MongoDB will be handled by Mongo JDBC if we specify on the query parameters (ref: [here](https://mongodb.github.io/mongo-java-driver/3.4/driver/tutorials/ssl/))

Reply:
@krishnapriawan 

1. What is the cause for indirect connections without SSL to fail? Is it the upgraded MongoDB driver?
2. The SSL on the JDBC is different, it corresponds to the checkbox that we have on the dashboard. This particular bit is so that the SSL certificate would always pass the connectivity (as it wouldn't verify with CA).